src/program/services/scrapers/base.py

```
@@ -75,10 +75,10 @@ def get_stremio_identifier(item: MediaItem) -> Tuple[str | None, str, str]:
-        with session.get(url, stream=True) as r:
-            r.raise_for_status()
-            torrent_data = r.content
-            torrent_dict = decode(torrent_data)
-            info = torrent_dict[b"info"]
-            infohash = hashlib.sha1(encode(info)).hexdigest()
+        r = session.get(url)
+        r.raise_for_status()
+        torrent_data = r.content
+        torrent_dict = bencodepy.decode(torrent_data)
+        info = torrent_dict[b"info"]
+        infohash = hashlib.sha1(bencodepy.encode(info)).hexdigest()
```

Ruff (0.14.2)

Probable use of insecure hash functions in `hashlib`: `sha1`

(S324)